contrib/epee/src/hex.cpp

```diff
@@ -83,4 +83,28 @@ namespace epee
   {
     return write_hex(out, src);
   }
+
+  std::vector<uint8_t> from_hex::vector(const std::string &src)
+  {
+    std::vector<uint8_t> result;
+
+    // erase possible previous errors
+    errno = 0;
+
+    // iterate over all the input characters
+    for (size_t i = 0; i < src.size(); i += 2) {
+      // skip spacing and colons
+      if (std::isspace(src.data()[i]) || src.data()[i] == ':')
+        ++i;
```

Comment:
This ignores an odd size's last digit, parses one value for ":", also for ":::::". errno doesn't get set here for 
":::askjndkj", which gets me 0 a 0 0 0. strtol (you might want strtoul instead btw) does not set errno for short read, it just returns.